password/impl/src/test/java/org/wildfly/security/password/impl/SunUnixMD5CryptTest.java

```diff
@@ -43,42 +42,31 @@
 public class SunUnixMD5CryptTest {
 
     @Test
-    public void testParseCryptStringWithoutRounds() throws NoSuchAlgorithmException, InvalidKeySpecException {
-        String cryptString = "$md5$zrdhpMlZ$$wBvMOEqbSjU.hu5T2VEP01";
-
-        // Get the spec by parsing the crypt string
-        SunUnixMD5CryptPassword password = (SunUnixMD5CryptPassword) ModularCrypt.decode(cryptString);
-        assertEquals(0, password.getIterationCount());
-
-        // Use the spec to build a new crypt string and compare it to the original
-        assertEquals(cryptString, ModularCrypt.encodeAsString(password));
+    public void testParseCryptStringWithoutRounds() throws InvalidKeySpecException {
+        testParseCryptString("$md5$zrdhpMlZ$$wBvMOEqbSjU.hu5T2VEP01", 0);
     }
 
     @Test
-    public void testParseCryptStringWithRounds() throws NoSuchAlgorithmException, InvalidKeySpecException {
-        String cryptString = "$md5,rounds=1000$saltstring$$1wGsmnKgDGdu03LxKu0VI1";
-
-        // Get the spec by parsing the crypt string
-        SunUnixMD5CryptPassword password = (SunUnixMD5CryptPassword) ModularCrypt.decode(cryptString);
-        assertEquals(1_000, password.getIterationCount());
-
-        // Use the spec to build a new crypt string and compare it to the original
-        assertEquals(cryptString, ModularCrypt.encodeAsString(password));
+    public void testParseCryptStringWithRounds() throws InvalidKeySpecException {
+        testParseCryptString("$md5,rounds=1000$saltstring$$1wGsmnKgDGdu03LxKu0VI1", 1_000);
     }
 
     @Test
-    public void testParseCryptStringWithBareSalt() throws NoSuchAlgorithmException, InvalidKeySpecException {
-        String cryptString = "$md5,rounds=1500$saltstring$F9DNxgHVXWaeLS9zUaWXd.";
+    public void testParseCryptStringWithBareSalt() throws InvalidKeySpecException {
+        testParseCryptString("$md5,rounds=1500$saltstring$F9DNxgHVXWaeLS9zUaWXd.", 1_500);
+    }
 
-        // Get the spec by parsing the crypt string
+    public void testParseCryptString(String cryptString, int iterCount) throws InvalidKeySpecException {
```

Comment:
Since this is just a helper test method, we can make this `private static`.

Reply:
Fixed!